src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -818,7 +818,7 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
             std::string ltype = ASRUtils::type_to_str(ASRUtils::expr_type(left));
             std::string rtype = ASRUtils::type_to_str(ASRUtils::expr_type(right));
             diag.add(diag::Diagnostic(
-                "Not Implemented: type mismatch in binary operator; only Integer, Real, Complex,"
+                "Type mismatch in binary operator; only Integer, Real, Complex,"
                 " Logical combinations and string concatenation/repetition are implemented for now.",
```

Comment:
How about just this?
```suggestion
                "Type mismatch in binary operator; the types must be compatible",
```
I think we now implement enough combinations, that if something is not supported, we should simply fix it, or provide an error message just for the "not implemented" stuff, but a general message should be simple I think.

Reply:
That's a great suggestion @certik. The suggested error message is more concise and simple.